offload/test/mapping/prelock.cpp

```diff
@@ -2,7 +2,7 @@
 // RUN: %libomptarget-run-generic %fcheck-generic
 
 // REQURIES: gpu
-// UNSUPPORTED: nvidia-gpu
+// UNSUPPORTED: nvidiagpu
 // UNSUPPORTED: amdgcn-amd-amdhsa
```

Comment:
Seems weird that we're doing `nvidia-gpu` instead of `nvptx64-nvidia-cuda` or `just `nvptx` when there's `amdgcn` right below it. I don't think there's a particular need to distinguish between the NVPTX triple and the NVPTX triple on a GPU.

Reply:
I'm just not sure where this test is supposed to be run. The test is unsupported on host targets, `amdgcn-amd-amdhsa`, and NVIDIA targets. Ideally we would only include the platform its meant to run on, but I'm not sure what platform that is.

Reply:
I'm just not sure where this test is supposed to be run. The test is unsupported on host targets, `amdgcn-amd-amdhsa`, and NVIDIA targets. Ideally we would only include the platform its meant to run on, but I'm not sure what platform that is.

Reply:
bug49334.cpp is another test case where I did this. I wasn't sure which targets the test was meant to support, so I just kept in a lot of the unsupported directives.

Upstream main:
https://github.com/llvm/llvm-project/blob/aa9bf71ca4eac50f4dfad9bb7b302491265cb889/offload/test/offloading/bug49334.cpp#L8-L16

My PR:
https://github.com/llvm/llvm-project/blob/88b8437ccfe02e9c4911610d3bcc31985f345918/offload/test/offloading/bug49334.cpp#L8-L11

Are these tests meant to be disabled?

Reply:
Lol, that list pretty much means "no one". I would just keep `nvptx64-nvidia-cuda` but drop the `-LTO` bit since it's irrelevant AFAIK. (I think there's a single test case where it matters but that's just because I don't have compiler-rt yet)

Reply:
I think the test where that matters is assert.c:

https://github.com/llvm/llvm-project/blob/88b8437ccfe02e9c4911610d3bcc31985f345918/offload/test/libc/assert.c#L6-L8

If the test isn't meant to run anywhere yet, I think `// UNSUPPORTED: host, gpu` might be more explicit.

Reply:
Oh yeah I wanted to get rid of that as well and just make it use the `libc` one, but for now it's probably fin. Can probably just have `REQUIRES: lto` or something.